c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -180,6 +185,43 @@ struct AllocParams {
 
 } // namespace
 
+
+class CachingAllocatorConfig {
+public:
+  size_t max_split_size;  // largest "large" alloc, "oversize" is not splitable
+
+  CachingAllocatorConfig()
+  : max_split_size(SIZE_MAX)
+  {
+    const char *val = getenv("PYTORCH_CUDA_ALLOC_CONF");
+    if (val != NULL) {
+      std::string config(val);
+      size_t pos = 0;
+
+      while (pos < config.length()) {
```

Comment:
I look pretty dimly on manual string parsing code like this, because it is very easy to get wrong. A higher level approach would be much better.